lib/Epub/EpubList/EpubIndex.cpp

```diff
@@ -0,0 +1,119 @@
+#include "EpubIndex.h"
+
+static const char *TAG = "PUBINDEX";
+#define PADDING 20
+#define ITEMS_PER_PAGE 5
+
+void EpubIndex::next()
+{
+  // must be loaded as we need the information from the epub
+  if (!epub)
+  {
+    load();
+  }
+  state.selected_item = (state.selected_item + 1) % epub->get_toc_items_count();
+}
+
+void EpubIndex::prev()
+{
+  // must be loaded as we need the information from the epub
+  if (!epub)
+  {
+    load();
+  }
+  state.selected_item = (state.selected_item - 1 + epub->get_toc_items_count()) % epub->get_toc_items_count();
+}
+
+bool EpubIndex::load()
+{
+  ESP_LOGI(TAG, "load");
+
+  if (!epub || epub->get_path() != selected_epub.path)
+  {
+    renderer->show_busy();
+    delete epub;
+
+    epub = new Epub(selected_epub.path);
+    if (epub->load())
+    {
+      ESP_LOGI(TAG, "Epub index loaded");
+      return false;
+    }
+  }
+  return true;
+}
+
+// TODO - this is currently pretty much a copy of the epub list rendering
```

Comment:
Totally agree on this point, let's do it in an additional PR to keep this core update as simpler as possible

Reply:
https://github.com/atomic14/diy-esp32-epub-reader/issues/69